modules/phase_vcf/1.0/phase_vcf.smk

```diff
@@ -0,0 +1,176 @@
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    logger.warning(
+                '\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}'
+                "\n" f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m'
+                )
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    logger.warning(
+                '\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}'
+                "\n" f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m'
+                )
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section 
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["phase_vcf"]`
+CFG = op.setup_module(
+    name = "phase_vcf",
+    version = "1.0",
+    subdirectories = ["inputs", "clair3", "filter_clair3", "whatshap_phase_vcf","outputs"]
+)
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _promethion_input,
+    _filter_clair3,
+    _phase_vcf_all
+
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _promethion_input:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam.bai"
+    run:
+        op.absolute_symlink(input.bam, output.bam)
+        op.absolute_symlink(input.bai, output.bai)
+
+rule _clair3:
+    input:
+        bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{sample_id}.bam",
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    params: 
+        model = "/bin/models/" + CFG["clair3"]["model"],
```

Comment:
I already approved this, but forgot to ask about this. How does this work with Clair3 installed via Conda?

Reply:
I assume you're talking about the model selection? The user would need to select the model they want to use then in the shell command ${{CONDA_PREFIX}} obtains the path to the activated clair3 conda environment which is added as a prefix to the bin/models + CFG["clair3"]["model"] command. I tried the workaround in the vcf2maf module but it didn't work the hash wasn't the same as the activated conda environment and this way is more streamlined